draft-tsai-duration-00.xml

```diff
@@ -275,6 +275,16 @@ This example must instead be expressed as -PT1H2M3S</t>
 </section>
 </section>
 
+
+<section anchor="security-considerations" title="Security Considerations">
+
+<t>When parsing, adding up each unit value may result in numeric overflow.
+For example, if an implementation represents a duration as a signed 64-bit integer of nanoseconds,
+then parsing PT2562047H47M16S.854775808 should report an error since this would otherwise overflow to a very large negative duration.
+Such bugs could conceivably be used to circumvent permissions checks based on time.</t>
+
+</section>
```

Comment:
Do we want to say anything about the practically maximum possible duration to have some upper bound on the string length/size?

Reply:
Every language handles durations differently. For example:

- Go treats it as a simple `int64` of nanoseconds.
- Python does something clever, where it can represent both accurate and nominal durations. If you parse a nominal duration, it remembers that some units are in days, months, years, etc. and doesn't try to convert them. Furthermore, Python integers under the hood automatically scale to infinite width.
- Java uses a 64-bit integer for seconds and a 32-bit integer for nanoseconds.
- Protobuf `google.protobuf.Duration` is similar to Java.

Since implementations range from have some practical limit to no limit at all, I think we should be silent.

Reply:
Or more generally say that implementations MAY impose a ceiling and SHOULD document it?

Reply:
I like that. I'll put that in section 3.